### PR TITLE
Initial sanitization doc changes

### PR DESCRIPTION
**Pull request by jairuzu**

### CODE REVIEW

File: docs/concepts/sanitization.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Sanitization
+
+> I wish it were Santa-ization.  Who wouldn't like that?
+>
+> &#8212; Anonymous
+
+## Definition
+
+[Sanitization](https://en.wikipedia.org/wiki/HTML_sanitization) within the context of Bandicoot refers to cleaning an HTML string to remove potentially malicious content. Unsanitized HTML is vulnerable to [cross site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting) which is something you need to avoid in your code. Bandicoot does **NOT** automatically sanitize your HTML, however it provides a way for you to bring your own sanitizing function to the party.
+
```
Comment (dvnrsn) [inline docs/concepts/sanitization.md:10]: 🎉

File: docs/concepts/sanitization.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+that javascript will be executed during deserialization. This is a security risk known as
+[cross site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting).
+
+## What this means for you
+
+### Sanitization
+Since Bandicoot does **NOT** automatically sanitize incoming or outgoing HTML, it is up to you as the developer/consumer to make sure that sanitization happens. However, Bandicoot does expose a [sanitizeHTML prop](/components/rich-text-editor.md#props) on the [RichTextEditor component](/components/rich-text-editor.md) which allows you to provide a sanitization function.
+
+If a sanitization function is provided through the `sanitizeHTML` prop, Bandicoot will pass incoming html strings through this function before inserting the content into the DOM (see [note 1](/concepts/sanitization.md#notes)). If provided, Bandicoot will also pass outgoing html through the `sanitizeHTML` function after serialization (see [note 2](/concepts/sanitization.md#notes)). If a `sanitizeHTML` function is not provided, Bandicoot will simply pass through the *unsanitized* html string - which is **NOT RECOMMENDED**!
```
Comment (dvnrsn) [inline docs/concepts/sanitization.md]: html vs HTML inconsistency

Comment (dvnrsn) [reply docs/concepts/sanitization.md]: `Bandicoot will simply pass through the unsanitized HTML string`

Isn't it more accurate to say `Bandicoot will dangerouslySetInnerHTML with the unsanitized HTML`

File: docs/concepts/sanitization.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+<RichTextEditor
+  sanitizeHTML={myOwnSanitizeHtmlFunction}
+/>
+
+function myOwnSanitizeHtmlFunction(suspectHTML, actionType) {
+  // Implement your logic to sanitize 'suspectHTML' into 'sanitizedHTML'.
+  // If desired, key off of the 'actionType' string.
+  return sanitizedHTML;
+}
+```
```
Comment (dvnrsn) [inline docs/concepts/sanitization.md:54]: to @joeldenning's point, it'd be nice to have some real use cases for discriminating by `actionType`

File: docs/concepts/serialization.md [hunk 4]
```diff
@@ -23,11 +23,4 @@ customized with [`richTextContext.addSerializer()`](/context/rich-text-context.m
...
 
 ### Deserialization
-Bandicoot accomplishes deserialization via React's [dangerouslySetInnerHTML](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml).
-React calls it dangerous for good reason -- if the HTML string has a `<script>` element or other javascript embedded into the dom elements,
-that javascript will be executed during deserialization. This is a security risk known as
-[cross site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting).
-
-Cross site scripting is something you need to avoid in your code. To do so, the HTML string needs to be
-[sanitized](https://en.wikipedia.org/wiki/HTML_sanitization). Bandicoot does not currently automatically sanitize your HTML
-before using dangerouslySetInnerHTML. However, [RichTextEditor](/components/rich-text-editor.md) does accept a prop `sanitizeHTML` that will be invoked on any HTML that is inserted into the DOM. Please use a library like [sanitize-html](https://github.com/punkave/sanitize-html) in the creation of the sanitization function you provide `sanitizeHTML`.
```
Comment (dvnrsn) [inline docs/concepts/serialization.md:33]: I think we should keep the part about recommending using an external library even if we don't actually name any.

Comment (jairuzu) [reply docs/concepts/serialization.md:33]: Added, see line 25 of docs/concepts/sanitization.md.

File: docs/concepts/sanitization.md (hunk 3 above)
Comment (jairuzu) [reply docs/concepts/sanitization.md:54]: Add some additional points in the comments to `function myOwnSanitizeHtmlFunction...`

File: docs/concepts/sanitization.md (hunk 2 above)
Comment (jairuzu) [reply docs/concepts/sanitization.md]: updated html vs HTML inconsistency.  Also updated the text to include a bit more clarification on the dangerouslySetInnerHTML stuff.

**Review by dvnrsn — APPROVED** [review]

**Pull request merged**